### PR TITLE
[Hub Generated] Publish private branch 'release-azurestackhci-Microsoft.AzureStackHCI-2021-09-01-preview'

### PR DESCRIPTION
**Pull request by pathuang-sia**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/published%2Fpathuang-sia%2Fazure-rest-api-specs-pr%2Frelease-azurestackhci-Microsoft.AzureStackHCI-2021-09-01-preview...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by pathuang-sia** [comment]
The PR is created based on the updates in the private branch. The updates in the PR has already been reviewed and approved with this PR [Azure/azure-rest-api-specs-pr/9073](https://github.com/Azure/azure-rest-api-specs-pr/pull/9073)

**Comment by pathuang-sia** [comment]
@xiaoxuqi-ms would you mind merging this given the approval in the approval here?
https://github.com/Azure/azure-rest-api-specs-pr/pull/9073

**Comment by xiaoxuqi-ms** [comment]
please fix the linter errors, model validation errors and prettier check. Other update LGTM.

**Comment by pathuang-sia** [comment]
> please fix the linter errors, model validation errors and prettier check. Other update LGTM.

Hi @xiaoxuqi-ms thanks for reviewing. The lintdiff, modelvalidation and prettier check failures we will be taking in our next version as mentioned in the private PR this PR was based off of: https://github.com/Azure/azure-rest-api-specs-pr/pull/9073#issuecomment-1276784267
Let me know if you have additional questions, happy to answer them.

File: custom-words.txt [hunk 1]
```diff
@@ -819,6 +819,7 @@ Fulltext
 fullyqualified
 functionappsettings
 galician
+galleryimage
```
Comment (TimLovellSmith) [inline custom-words.txt:822]: You shouldn't need to add this. Fix capitalization elsewhere in the PR so that spelling check passes without adding this.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHCI/galleryimages/{galleryimagesName}": {
```
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json:40]: >galleryimages [](http://example.com/codeflow?start=106&length=13)

should be capitalized 'galleryImages' in all paths

Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json:40]: >galleryimagesName [](http://example.com/codeflow?start=121&length=17)

should be 'galleryImageName' or 'imageName'

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Retrieves the galleryimages resource.",
```
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json:71]: >Retrieves [](http://example.com/codeflow?start=28&length=9)

The tense of the sentence doesn't make sense. 200 isn't going to do *retrieves*, it is an already successful response so it indicates the resource has been *retrieved*

Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json:71]: > the galleryimages resource." [](http://example.com/codeflow?start=37&length=29)

Docs should talk about 'the gallery image resource' in plain English style

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+            "schema": {
+              "$ref": "#/definitions/galleryimages"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "galleryimages_CreateOrUpdate",
```
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json:79]: >galleryimages_CreateOrUpdate [](http://example.com/codeflow?start=24&length=28)

OperationIds should be capitalizing GalleryImages or galleryImages, I forget which.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json [hunk 5]
```diff
@@ -0,0 +1,590 @@
...
+        "nextLink": {
+          "type": "string",
+          "description": "Link to the next set of results.",
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
+    "galleryimagesNameParameter": {
```
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json:579]: >galleryimagesNameParameter [](http://example.com/codeflow?start=5&length=26)

suggest calling it GalleryImageNameParameter

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json [hunk 6]
```diff
@@ -0,0 +1,590 @@
...
+        "publisher": {
+          "type": "string",
+          "description": "The name of the gallery image definition publisher."
+        },
+        "offer": {
+          "type": "string",
+          "description": "The name of the gallery image definition offer."
+        },
+        "sku": {
+          "type": "string",
```
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json:447]: Shouldn't you enumerate or document allowed values?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/virtualHardDisks.json [hunk 7]
```diff
@@ -0,0 +1,470 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHCI/virtualharddisks/{virtualharddisksName}": {
```
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/virtualHardDisks.json:40]: >virtualharddisks [](http://example.com/codeflow?start=106&length=16)

all resource type names should be camelCase

**Review by xiaoxuqi-ms — APPROVED** [review]

**Review by xiaoxuqi-ms — APPROVED** [review]

**Comment by xiaoxuqi-ms** [comment]
@pathuang-sia , while we are reviewing this pr, new update are merged into 2021-07-01-preview, in pr: https://github.com/Azure/azure-rest-api-specs/pull/18152. so there are breaking changes. Please help to confirm if the breaking changes are introduced intentionally or should these changes be synced to 2021-09-01-preview to avoid these breaking changes.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/virtualHardDisks.json (hunk 7 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/virtualHardDisks.json]: won't fix, will take in next version

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/marketplaceGalleryImages.json [hunk 8]
```diff
@@ -0,0 +1,587 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHCI/marketplacegalleryimages/{marketplacegalleryimagesName}": {
```
[the remark this answers is not in the document]
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/marketplaceGalleryImages.json]: won't fix, will take in next version

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json (hunk 5 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json]: Fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json (hunk 3 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json]: fixed

Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json]: fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json (hunk 2 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json]: fixed

File: custom-words.txt (hunk 1 above)
Comment (pathuang-sia) [reply custom-words.txt]: fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json (hunk 2 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json]: won't fix, will take in the next version

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json (hunk 4 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-09-01-preview/galleryImages.json:79]: won't fix, will take in next version

**Comment by xiaoxuqi-ms** [comment]
I think the first letter of the property name can be lower-case: for example, galleryimagesProperties is fine. But not GalleryimagesProperties
